endpoints to fetch a list of all pending or tentative/accepted talks for the speaker

mgerhold opened the pull request:
This PR adds two new endpoints:
- `GET api/dashboard/speaker/pending-talks`: Requires the speaker role, returns an array of all pending (i.e. non-approved) talks or the currently logged in speaker. The returned data has the following structure:
  ```json5
  {
    "pending_talks": [
      {
        "id": 10,
        "event_id": 1,
        "title": "Programmieren wie ein Profi",
        "description": "In diesem Vortrag lernst du, wie du professionell programmierst.",
        "notes": "Ich kann nur am zweiten Tag der Konferenz.",
        "requested_changes": null,
        "created_at": "2025-01-11 13:53:15",
        "speaker": {
          "id": 2,
          "user_id": 1,
          "name": "coder2k",
          "short_bio": "Test-Conf Host, Software-Entwickler, freier Dozent, Twitch-Partner",
          "bio": "Michael (coder2k) hat vor über 20 Jahren 'Turbo Pascal und Delphi für Kids' gelesen und sich seitdem mit dem Programmieren in verschiedenen Programmiersprachen beschäftigt. Er ist tätig als Software-Entwickler im Embedded-Umfeld und freier Dozent. Seit drei Jahren programmiert er auch auf Twitch und ist seit Anfang 2024 Twitch-Partner. Michael ist es wichtig, Wissen mit anderen auszutauschen und sich dadurch gemeinsam weiterzuentwickeln und neue Dinge zu lernen – und daraus ist auch die Idee zur Test-Conf entstanden.",
          "photo": "images/coder2k.jpg",
          "photo_mime_type": "image/jpeg",
          "is_approved": true,
          "visible_from": "2024-06-01 15:00:00",
          "requested_changes": null
        },
        "tags": [
          {
            "color_index": 4,
            "text": "Programmierung"
          },
          {
            "color_index": 1,
            "text": "Maker"
          },
          {
            "color_index": 2,
            "text": "Hacking"
          }
        ],
        "possible_durations": [
          30,
          45,
          60
        ],
        "suggested_time_slot": null
      }
      // more talks...
    ]
  }
  ```

- `GET api/dashboard/speaker/tentative-or-accepted-talks/(:num)`, where `:num` is an event id: Requires the speaker role, returns an array of all tentative (i.e. approved, but without an accepted time slot) and accepted (i.e. approved with accepted time slot) events for the specified event for the currently logged in user. The returned data has the same structure as above, but with the additional field `time_slot_accepted` of type `bool`.
  Possible errors:
  - `EVENT_NOT_FOUND` (404)

Additionally, the endpoint `GET api/dashboard/admin/tentative-talks` has been changed to `GET api/dashboard/admin/tentative-or-accepted-talks/(:num)`. It now requires passing an event ID and it returnes not only tentative, but also accepted talks. The structure of the data is the same as for `GET api/dashboard/speaker/tentative-or-accepted-talks/(:num)`.

PurpurTentakel97 commented on app/Controllers/Talk.php:
```diff
@@ -154,6 +154,29 @@ public function submit(): ResponseInterface
+        $pendingTalks = $talkModel->getPendingForSpeaker($this->getLoggedInUserId());
+
+        $pendingTalks = $this->addAdditionalDataToTalks($pendingTalks);
+        return $this->response->setJSON(['pending_talks' => $pendingTalks])->setStatusCode(200);
```
please remove the `pending_talks` tag so that the array is the top element.

And for some reason the `suggested_time_slot` has CamelCase Keys.
please change that into `snake_case`

```
            "suggested_time_slot": {
                "id": 2,
                "eventId": 1,
                "startTime": "2024-06-22 14:00:00",
                "duration": 45,
                "isSpecial": false
            }
```

mgerhold replied:
Yes, will do 👍

PurpurTentakel97 commented on app/Controllers/Talk.php:
```diff
@@ -154,6 +154,29 @@ public function submit(): ResponseInterface
+        $tentativeTalks = $talkModel->getTentativeOrAcceptedForSpeaker($this->getLoggedInUserId(), $eventId);
+
+        $tentativeTalks = $this->addAdditionalDataToTalks($tentativeTalks);
+        return $this->response->setJSON(['tentative_talks' => $tentativeTalks])->setStatusCode(200);
```
same here I think

mgerhold commented:
I removed the wrapping JSON objects and replaced them with plain JSON arrays. I also fixed the camelCase capitalization in `suggested_time_slot` (now it's snake_case, like all the other stuff).

mgerhold commented:
fixes #123
